src/poetry/installation/installer.py

```diff
@@ -291,12 +291,10 @@ def _do_install(self) -> int:
         lockfile_repo = LockfileRepository()
         self._populate_lockfile_repo(lockfile_repo, ops)
 
-        if self._update:
+        if self._lock and self._update:
```

Comment:
It would make sense, however it's not true, since you can run `poetry lock --no-update`, which ends up in `self._lock = True` and `self._update = False`.

Relevant extracts below from **commands/lock.py** and **installer.py**

https://github.com/python-poetry/poetry/blob/161b19cb4e4686fc5a0a7925001534a87f6c4052/src/poetry/console/commands/lock.py#L55

https://github.com/python-poetry/poetry/blob/161b19cb4e4686fc5a0a7925001534a87f6c4052/src/poetry/installation/installer.py#L153-L161

Reply:
Perhaps will be clearer if I swap the conditions?

```suggestion
        if self._update and self._lock:
```

Reply:
Arghh, too many options and a confusing mapping. I always forget one or the other. Good catch! 👍

I don't think the order makes a difference.